cql3/restrictions/statement_restrictions.hh

```diff
@@ -414,7 +414,11 @@ public:
     }
 
     bool ck_restrictions_need_filtering() const {
-        return _partition_key_restrictions->has_unrestricted_components(*_schema) || _clustering_columns_restrictions->needs_filtering(*_schema);
+        return _partition_key_restrictions->has_unrestricted_components(*_schema) 
```

Comment:
I can't write a comment on the commit message, so I'll write it here.
When fixing a problem, please don't just say you "handle it properly" or "fix it". Please also explain what was the bug earlier - what was actually wrong? Did such query print an error? Fail? Produce wrong results (if so, which sort of wrong results)?